sphinx_gallery/gen_gallery.py

```diff
@@ -688,6 +694,207 @@ def write_computation_times(gallery_conf, target_dir, costs):
+                                for level2 in range(level + 2)]):
+                            leaf_count += 1
+                    node_to += f'\n({leaf_count})'
+                    lut[struct[level + 1]] = node_to
+                    if leaf_count > 10:
+                        color = 'red'
+                    elif leaf_count > 5:
+                        color = 'orange'
+                    else:
+                        color = 'yellow'
+                    dg.attr('node', color=color)
+                else:
+                    dg.attr('node', color='lightblue2')
+                dg.node(node_to)
+                dg.edge(node_from, node_to)
+        # add modules with all API entries
+        dg.attr('node', color='lightblue2')
+        for module in gallery_conf['api_entries']['module']:
+            struct = module.split('.')
+            for i in range(len(struct) - 1):
+                if struct[i + 1] not in lut:
+                    dg.edge(struct[i], struct[i + 1])
+    else:
+        assert isinstance(entries, dict)
+        for entry, refs in entries.items():
+            dg.attr('node', color='lightblue2')
+            dg.node(entry)
+            dg.attr('node', color='yellow')
+            for ref in refs:
+                # remove sphx_glr_auto_xxx
+                ref = '_'.join(ref.split('_')[3:])
+                dg.node(ref)
+                dg.edge(entry, ref)
+
+    dg.attr(overlap='scale')
+    dg.save(fname)
+
+
+def write_api_entry_usage(app, docname, source):
+    gallery_conf = app.config.sphinx_gallery_conf
+    # since this is done at the gallery directory level (as opposed
+    # to in a gallery directory, e.g. auto_examples), it runs last
+    # which means that all the api entries will be in gallery_conf
+    if 'sg_api_usage' not in docname or \
+            'api_entries' not in gallery_conf or \
+            gallery_conf['backreferences_dir'] is None:
+        return
+    backreferences_dir = os.path.join(gallery_conf['src_dir'],
+                                      gallery_conf['backreferences_dir'])
+
+    example_files = set.union(
+        *[gallery_conf['api_entries'][obj_type]
+          for obj_type in ('class', 'method', 'function')
+          if obj_type in gallery_conf['api_entries']])
+
+    total_count = len(example_files)
+    if total_count == 0:
+        return
+
+    def get_entry_type(entry):
+        if entry in gallery_conf['api_entries']['class']:
+            return 'class'
+        elif entry in gallery_conf['api_entries']['method']:
+            return 'meth'
+        else:
+            assert entry in gallery_conf['api_entries']['function']
+            return 'func'
+
+    # find used and unused API entries
+    unused_api_entries = list()
+    used_api_entries = dict()
+    for entry in example_files:
+        # don't include built-in methods etc.
+        if re.match(gallery_conf['missing_doc_ignore'],
+                    entry.split('.')[-1]) is not None:
```

Comment:
I don't think you should only look at the method name (i.e., operate on `split('.')[-1]). People can split on the last `.` in regex if they really want to, e.g., something like `\.__[^.]*__$` does this (or is close to it) for `__.*__` method names.

Reply:
Agreed